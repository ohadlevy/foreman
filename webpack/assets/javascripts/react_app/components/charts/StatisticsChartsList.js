import React from 'react';
import StatisticsChartBox from './StatisticsChartBox';
import styles from './StatisticsChartsListStyles';

const StatisticsChartsList = ({data}) => {
    let charts = [];

    data.forEach(chart => {
      charts.push(<StatisticsChartBox key={chart.id} {...chart} />);
    });

    return (
      <div className="collection" style={styles.root}>
        {charts}
      </div>
    );
};

export default StatisticsChartsList;
