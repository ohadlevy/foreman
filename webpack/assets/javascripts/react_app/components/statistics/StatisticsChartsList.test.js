import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import React from 'react';

import StatisticsChartsList from './StatisticsChartsList';
import { statisticsData } from './StatisticsChartsList.fixtures';

describe('StatisticsChartsList', () => {
  it('should render no panels for empty data', () => {
    const wrapper = shallow(
      <StatisticsChartsList charts={[]} data={statisticsData} />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render two panels for fixtures data', () => {
    const wrapper = shallow(
      <StatisticsChartsList charts={statisticsData} data={statisticsData} />
    );

    expect(wrapper.render().find('.chart-box')).toHaveLength(2);
  });
});
