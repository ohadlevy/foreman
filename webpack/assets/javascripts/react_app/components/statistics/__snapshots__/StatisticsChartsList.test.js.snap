// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticsChartsList should render no panels for empty data 1`] = `
<div
  className="statistics-charts-list-root cards-pf"
>
  0
</div>
`;
